pkg/argocd/update.go

```diff
@@ -288,6 +288,18 @@ func UpdateApplication(updateConf *UpdateConfiguration, state *SyncIterationStat
 		if needsUpdate(updateableImage, applicationImage, latest) {
 
 			imgCtx.Infof("Setting new image to %s", applicationImage.WithTag(latest).GetFullNameWithTag())
+
+			// Check if new image is alredy set in Application Spec when write back is set to argocd
+			// and compare with new image
+			appImage, err := getAppImage(&updateConf.UpdateApp.Application, applicationImage.WithTag(latest))
+			if err != nil {
+				continue
+			}
+			if appImage == applicationImage.WithTag(latest).GetFullNameWithTag() {
+				imgCtx.Infof("New image %s already set in spec", applicationImage.WithTag(latest).GetFullNameWithTag())
+				continue
+			}
+
 			needUpdate = true
 
 			err = setAppImage(&updateConf.UpdateApp.Application, applicationImage.WithTag(latest))
```

Comment:
`applicationImage.WithTag(latest)` and `applicationImage.WithTag(latest).GetFullNameWithTag()` are called a few times, maybe we should extract into variables to avoid computing it repeatedly.